### DISCUSSION TITLE
Can't build v0.160.4-pre on windows due "failed run custom command for `msvc_spectre_libs v0.1.2`"

### DISCUSSION THREAD

**Discussion by gohellp** [zed-industries/zed#20142] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

- git clone this repository
- `rustup update`
- `rustup target add wasm32-wasip1`
- Edit VS 2022 installation to add `MSVC v143 - VS 2022 C++ x64/x86 build tools` and `MSVC v143 - VS 2022 C++ x64/x86 Spectre-mitigated libs`
- Install Windows 11 SDK by url in docs/src/development/windows.md
- `cargo run`

I don't understand why this behavior happens a bit, even though I went through all the necessary steps to build the IDE. Most likely, the error is somewhere in the behavior of the system, but I had never encountered anything like this before, and even when I first built the IDE, when I was just installing all the necessary components, everything was able to work at once.

I'm sorry if I chose the wrong template, I just couldn't find anything that could fit this situation and chose the nearest one.

### Environment

Device name	ShadowPC
Processor	AMD Ryzen 5 3600 6-Core Processor                 3.60 GHz
Installed RAM	64.0 GB
Device ID	9294F0D3-7328-40DD-91AD-0C01E693ABF8
Product ID	00331-10000-00001-AA481
System type	64-bit operating system, x64-based processor
Pen and touch	No pen or touch input is available for this display

### If applicable, add mockups / screenshots to help explain present your vision of the feature

Proof of installed msvc_spectre_libs (I tried to change the language to English  for easier understanding but VS Installer doesn't care about that)
![image](https://github.com/user-attachments/assets/1ab2e5ec-555c-4fa8-806b-07ae7ec50773)

### If applicable, attach your Zed.log file to this issue.

<details><summary>build log</summary><pre>

The following warnings were emitted during compilation:

warning: msvc_spectre_libs@0.1.2: No spectre-mitigated libs were found. Please modify the VS Installation to add these.

error: failed to run custom build command for `msvc_spectre_libs v0.1.2`

Caused by:
  process didn't exit successfully: `E:\coding\Rust\zed\target\debug\build\msvc_spectre_libs-81177709191ad7b1\build-script-build` (exit code: 101)
  --- stdout
  cargo:warning=No spectre-mitigated libs were found. Please modify the VS Installation to add these.

  --- stderr
  thread 'main' panicked at C:\Users\gohel\.cargo\registry\src\index.crates.io-6f17d22bba15001f\msvc_spectre_libs-0.1.2\build.rs:37:13:
  No spectre-mitigated libs were found. Please modify the VS Installation to add these.
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
</pre></details>

**Comment by SomeoneToIgnore** [comment]
> IDE

Zed is not an IDE, it's an editor.

> не поддерживается

I believe that is your issue, why does it say "unsupported"?

My installation does not have this, saying `latest` instead and it works, same as it works in our CI and others' builds.

**Comment by gohellp** [comment]
@SomeoneToIgnore, I don't know, that's all I can install. VS 2022 is updated, so I don't know how to get the latest components.

I'm here to figure out how to fix this problem, and I'd like to create a non-issue bug report, but I haven't found a suitable template. So it would only be appreciated if you could remove the “defect” tag.

**Comment by SomeoneToIgnore** [comment]
Then I'd suggest breaking a habit of creating issues, as there's a whole section for this purpose where I'm moving this.

**Reply by gohellp** [reply]
Okay, thank you. I just don't really understand the GitHub culture yet

**Comment by gohellp** [comment]
Fixed by changing locale settings for VS Installer

**Reply by Zapelmon** [reply]
Can you answer how? I have an issue: "error: failed to run custom build command for msvc_spectre_libs v0.1.2". However, I installed VS Build Tools 17.12.1 and `MSVC v143 - VS 2022 C++ x64/x86 Spectre-mitigated libs`, which are the only ones available. (Между тем, судя по скриншоту, вы русскоязычный. В общем, я так понимаю, вы сменили локаль с русского на английский. Хотел бы узнать, как вы решили проблему.)

**Reply by gohellp** [reply]
You can find the required settings file at the following path:
`AppData\Roaming\Visual Studio Setup\user.json`
I just changed the `locale` parameter to `en-US`.

I assume there is a way to make it WAY easier. But at that point I just got tired of trying to fix it somehow and started looking for a settings file, driven by the evil possibility of fixing getting versions by simply changing the language (btw, I installed VS Installer on Win with the En locale, lol).

**Reply by Zapelmon** [reply]
btw, worked. thx.

**Reply by parz3val** [reply]
Thanks. Worked for me as well.  @gohellp

**Reply by picania** [reply]
В русской версии you had to search by word `spectre` and select this
<img width="662" alt="image_2025-01-13_00-25-19" src="https://github.com/user-attachments/assets/5457a96c-7988-4abf-b9a4-64d7ca3759b7" />
